nixos/tests/emacs-daemon.nix

```diff
@@ -0,0 +1,40 @@
+import ./make-test.nix ({ pkgs, ...} : {
+  name = "emacs-daemon";
+  meta = with pkgs.stdenv.lib.maintainers; {
+    maintainers = [ DamienCassou ];
+  };
+
+  enableOCR = true;
+
+  machine =
+    { config, pkgs, ... }:
+
+    { imports = [ ./common/user-account.nix ];
+
+      services.xserver.enable = true;
+
+      services.xserver.displayManager.gdm = {
+        enable = true;
+        autoLogin = {
+          enable = true;
+          user = "alice";
+        };
+      };
+      services.xserver.desktopManager.gnome3.enable = true;
+      environment.gnome3.packageSet = pkgs.gnome3_18;
+
+      virtualisation.memorySize = 512;
+      services.emacs.enable = true;
+    };
+
+  testScript =
+    ''
+      $machine->start();
+      $machine->waitForX;
+      $machine->succeed("su - alice -c 'DISPLAY=:0.0 emacs -Q &'");
+      $machine->succeed("xauth merge ~alice/.Xauthority");
+      $machine->waitForWindow(qr/emacs/);
```

Comment:
Try `$machine->waitForWindow(qr/emacs\@machine/);`

Seems like OCR is having trouble with the monospace font and/or red buffer text. `$machine->waitForText("GNU Emacs");` works better.

Also it might be simpler to run the test as root, like in the firefox test.

Unfortunately the `waitForUnit` method is for system services not user services, so we can't easily test the service.